src/containers/Tutorials/WatchPlaylist/index.tsx

```diff
@@ -0,0 +1,122 @@
+    if (playlist?.video_list.length) {
+      const index = playlist.video_list.findIndex((video) => video.uuid === currentVideo?.uuid);
+      if (index !== -1 && index !== playlist.video_list.length - 1) {
+        setCurrentVideo(playlist.video_list[index + 1]);
+      }
+    }
+  }, [currentVideo, playlist]);
+
+  const renderVideoList = (): ReactNode => (
+    <div className="WatchPlaylist__list">
+      <div className="WatchPlaylist__list-header">
+        <h5 className="WatchPlaylist__list-heading">{playlist?.title}</h5>
+        <div className="WatchPlaylist__list-length">{playlist?.video_list.length} videos</div>
+      </div>
+      <div className="WatchPlaylist__list-body">
+        {playlist?.video_list.map((video, index) => (
+          <div
+            className={clsx(
+              'WatchPlaylist__list-video',
+              currentVideo?.uuid === video.uuid && 'WatchPlaylist__list-video--active',
+            )}
+            key={video.video_id}
+            onClick={() => setCurrentVideo(video)}
+            role="button"
+            tabIndex={0}
+            title={video.title}
+          >
+            <div className="WatchPlaylist__list-video-number">
+              {video.uuid === currentVideo?.uuid ? <Icon icon={IconType.play} /> : index + 1}
+            </div>
+            <div className="WatchPlaylist__list-video-details">
+              <div className="WatchPlaylist__list-video-top">{video.title}</div>
+              <div className="WatchPlaylist__list-video-bottom">
+                <span className="WatchPlaylist__list-video-author">{video.author}</span>
+                <span className="WatchPlaylist__list-video-duration">
+                  {getFormattedTime(video.duration, TimeFormat.digital)}
+                </span>
+              </div>
+            </div>
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+
+  if (loading)
+    return (
+      <div className="WatchPlaylist__loader">
+        <Loader />
+      </div>
+    );
+  if (errorMessage) return <div className="WatchPlaylist__error">{errorMessage}</div>;
+  if (!playlist || !currentVideo) return <EmptyPage />;
+  return (
+    <div className="WatchPlaylist">
+      <div className="WatchPlaylist__main">
+        <VideoPlayer
+          className="WatchPlaylist__video"
+          onEnded={handleVideoEnd}
+          source={playlist.playlist_type}
+          videoId={currentVideo.video_id}
+        />
+        <div className="WatchPlaylist__video-details">
+          <h4 className="WatchPlaylist__video-title"> {currentVideo.title} </h4>
+          <div className="WatchPlaylist__video-date">
+            Date Published: {format(parseISO(currentVideo.published_at), 'MMM dd, yyyy')}
+          </div>
+          <div className="WatchPlaylist__video-author">
+            {playlist.playlist_type === Source.youtube ? 'Youtube' : 'Vimeo'} Channel:{' '}
+            <span className="WatchPlaylist__video-author-name">{currentVideo.author}</span>
```

Comment:
Yeah, thought of that but since the Figma file didn't have the external link icon I didn't add it. 🤔 

hmm, it's a good idea.

Reply:
@jamessspanggg about this

we're not storing the channel URL rn. I could use the author name to generate the URL, but it won't be guaranteed it will be correct. So, I am leaving this for now.

Reply:
noted on this, maybe something to be added in the near future.

Reply:
I think can request from the BE to add the channel URL as it feels slightly weird from the user's perspective as the coloured text indicates an action that can be taken